redbot/core/events.py

```diff
@@ -241,10 +239,112 @@ async def on_command_error(ctx, error, unhandled_by_cog=False):
+                    message = _(
+                        "Argument `{parameter_name}` must be an integer between {minimum} and {maximum}."
+                    )
+            elif isinstance(error.value, float):
+                if (error.minimum == 0 or error.minimum == 1) and error.maximum is None:
+                    message = _("Argument `{parameter_name}` must be a positive number.")
+                elif error.minimum is None and error.maximum is not None:
+                    message = _(
+                        "Argument `{parameter_name}` must be a number no more than {maximum}."
+                    )
+                elif error.minimum is not None and error.maximum is None:
+                    message = _(
+                        "Argument `{parameter_name}` must be a number no less than {maximum}."
+                    )
+                elif error.maximum is not None and error.minimum is not None:
+                    message = _(
+                        "Argument `{parameter_name}` must be a number between {minimum} and {maximum}."
+                    )
+            else:
+                if error.minimum is None and error.maximum is not None:
+                    message = _(
+                        "Argument `{parameter_name}` must be a string with a length of no more than {maximum}."
+                    )
+                elif error.minimum is not None and error.maximum is None:
+                    message = _(
+                        "Argument `{parameter_name}` must be a string with a length of no less than {minimum}."
+                    )
+                elif error.maximum is not None and error.minimum is not None:
+                    message = _(
+                        "Argument `{parameter_name}` must be a string with a length of between {minimum} and {maximum}."
+                    )
+            await ctx.send(
+                message.format(
+                    maximum=error.maximum,
+                    minimum=error.minimum,
+                    parameter_name=ctx.current_parameter.name,
+                )
+            )
+            return
         elif isinstance(error, commands.BadArgument):
+            if isinstance(converter, commands.Range):
+                if converter.annotation is int:
+                    if (converter.min == 0 or converter.min == 1) and converter.max is None:
+                        message = _("Argument `{parameter_name}` must be a positive integer.")
+                    elif converter.min is None and converter.max is not None:
+                        message = _(
+                            "Argument `{parameter_name}` must be an integer no more than {maximum}."
+                        )
+                    elif converter.min is not None and converter.max is None:
+                        message = _(
+                            "Argument `{parameter_name}` must be an integer no less than {maximum}."
```

Comment:
```suggestion
                            "Argument `{parameter_name}` must be an integer no less than {minimum}."
```
Here, we check if the converter's max is None, so it will always read "no less than None", instead of our desired result. It just requires changing this part of the string in the suggestion.